core/remote/remotemodelserver.cpp

```diff
@@ -438,6 +478,7 @@ void RemoteModelServer::layoutChanged()
 void RemoteModelServer::layoutChanged(const QList<QPersistentModelIndex> &parents,
                                       QAbstractItemModel::LayoutChangeHint hint)
 {
+    m_pendingDataChanged.clear();
```

Comment:
Are you sure this isn't too aggressive? IIRC the client keeps some data outside of the affected areas, which would potentially show outdated information if we drop the changes here, no?

A safer alternative might be to send all pending changes at this point.